doc/userdoc/contribute/templates_styleguides/styleguide.rst

```diff
@@ -0,0 +1,608 @@
+                    &=  a^2 + 2ab + b^2
+
+For in-text math, use the math role::
+
+   Now we can see :math:`x=1` for this example.
+
+These will be rendered as
+
+   Now we can see :math:`x=1` for this example.
+
+Admonitions
+~~~~~~~~~~~
+
+Admonitions are directives that render as highlighted blocks to draw the reader's attention to a particular point.
+
+Use them sparingly.
+
+
+Use the admonition
+
+* "See also" to reference internal or external links (only in cases where the reference should stand out),
+
+* "Note" to add additional information that the reader needs to be aware of,
+
+* "Warning" to indicate that something might go wrong without the provided information, and
+
+* "Danger" if the situation may cause severe, possibly irreversible, problems.
+
+
+If you want a custom admonition use::
+
+   .. admonition:: custom label
+
+         Here is some text
+
+Rendered as
+
+
+   .. admonition:: custom label
+
+         Here is some text
+
+
+References
+~~~~~~~~~~
+
+For referencing reStructuredText files within the documentation, use the ``:doc:`` role. It requires the relative path to
+the file::
+
+   :doc:`path/to/file`
+
+In this case, the link caption will be the title of the given document.
+
+You can specify the text you want to use for the link by doing the following::
+
+   :doc:`custom label <path/file>`
+
+This will be rendered as
+
+   :doc:`Top header of file`
+
+   :doc:`custom label <file>`
+
+For cross-referencing specific section headings, figures, or other arbitrary places within file, use the ``:ref:`` role.
+
+The ``:ref:`` role requires a reference label that looks like this ``.. _ref-label:``. Each reference label must be unique
+in the documentation. Separate the reference label from the text it is referecing with a new line::
+
+   .. _my-ref-label:
+
+   Section to cross-reference
+   --------------------------
+
+   Some content in this section.
+   It includes the cross-referecing role :ref:`my-ref-label`.
+
+
+To reference figures or arbitrary places in a file, you must include a custom
+label in the reference for it to work::
+
+    :ref:`custom label <my-arbitrary-place-label>`
+
+
+
+Special links
+~~~~~~~~~~~~~
+
+.. attention::
+
+  The items in this section are still in development and have not been incorporated into nest:master!
+
+Links to functions
+^^^^^^^^^^^^^^^^^^
+
+To link PyNEST API functions used in-text to the API reference page use the following syntax::
+
+   :py:func:`.Create`
+
+Rendered as
+
+   :py:func:`.Create`
+
+Classes, methods etc. can also be linked: ``:py:class:`` ``:py:meth:``.
+If you want to explictly show a complete function call, like ``nest.Create("iaf_psc_apha")``, the link cannot be used.
+
+.. note::
+
+   Functions within classes NodeCollection and SynapseCollection require different syntax as follows
+
+   ``:py:func:`~nest.lib.hl_api_types.SynapseCollection.funcname```
+
+   ``:py:func:`~nest.lib.hl_api_types.NodeCollection.funcname```
```

Comment:
Could you explain why?

Reply:
It turns out I was mistaken as I  assumed get() and set() were functions but they are actually methods. Still, because they are both used in two classes, they have slightly special markup. I added text explaining this.